1-Calling-MSGraph/1-1-AzureAD/README.md

```diff
@@ -146,73 +146,84 @@ Start the application, sign-in and check the result in the console.
 {
-    var accounts = await app.GetAccountsAsync();
-    result = await app.AcquireTokenSilent(scopes, accounts.FirstOrDefault())
-                .ExecuteAsync();
-}
-catch (MsalUiRequiredException)
-{
-    result = await app.AcquireTokenInteractive(scopes)
-                .ExecuteAsync();
+   string authority = string.Concat(configuration.Instance, configuration.TenantId);
+
+   // Initialize the MSAL library by building a public client application
+   application = PublicClientApplicationBuilder.Create(configuration.ClientId)
+                                             .WithAuthority(authority)
+                                             .WithDefaultRedirectUri()
+                                             .Build();
+
+   // We intend to obtain a token for Graph for the following scopes (permissions)
+   string[] scopes = new[] { "user.read" };
+
+   AuthenticationResult result;
+   try
+   {
+         var accounts = await application.GetAccountsAsync();
+         result = await application.AcquireTokenSilent(scopes, accounts.FirstOrDefault())
+         .ExecuteAsync();
+   }
+   catch (MsalUiRequiredException ex)
+   {
+         result = await application.AcquireTokenInteractive(scopes)
+         .WithClaims(ex.Claims)
+         .ExecuteAsync();
+   }
+
+   return result.AccessToken;
 }
 ```
 
-4- Instantiate `GraphServiceClient` (from [Microsoft.Graph NuGet package](https://docs.microsoft.com/graph/sdks/sdk-installation)) using the Microsoft Graph access token acquired in the previous step.
+3- The method **SignInAndInitializeGraphServiceClient** initializes the Graph SDK
 
 ```csharp
-private static GraphServiceClient GetGraphServiceClient(string accessToken, string graphApiUrl)
+ private async static Task<GraphServiceClient> SignInAndInitializeGraphServiceClient(PublicClientApplicationOptions configuration)
 {
-    GraphServiceClient graphServiceClient = new GraphServiceClient(graphApiUrl,
-                                                            new DelegateAuthenticationProvider(
-                                                                async (requestMessage) =>
-                                                                {
-                                                                    await Task.Run(() =>
-                                                                    {
-                                                                        requestMessage.Headers.Authorization = new AuthenticationHeaderValue("bearer", accessToken);
-                                                                    });
-                                                                }));
-
-    return graphServiceClient;
+   GraphServiceClient graphClient = new GraphServiceClient(MSGraphURL,
+         new DelegateAuthenticationProvider(async (requestMessage) =>
+         {
+            requestMessage.Headers.Authorization = new AuthenticationHeaderValue("bearer", await SignInUserAndGetTokenUsingMSAL(configuration));
```

Comment:
and define the scopes here and pass them to SignInUserAndGetTokenUsingMSAL ?

Reply:
@jmprieur , I've declared the scope in Main and passed it as parameter in following methods. Kindly review.
Thanks!